apps/wallet/src/screens/Authentication/Login/index.tsx

```diff
@@ -1,33 +1,76 @@
-import type { FC } from 'react';
+import { type FC, useState } from 'react';
 import { StyleSheet } from 'react-native';
-import { appState } from '@walless/engine';
+import { logger } from '@walless/core';
 import { Anchor, Text, View } from '@walless/gui';
-import { appActions } from 'state/app';
+import { useSafeAreaInsets, useSnapshot } from 'hooks';
+import { showError } from 'modals/Error';
+import { appState } from 'state';
 import assets from 'utils/assets';
-import { useSafeAreaInsets, useSnapshot } from 'utils/hooks';
+import {
+	checkInvitationCode,
+	signInWithGoogle,
+	signInWithTorusKey,
+	ThresholdResult,
+} from 'utils/auth';
+import type { UserAuth } from 'utils/firebase';
+import { navigate } from 'utils/navigation';
 
 import SignInHeader from './SignInHeader';
 import SignInInner from './SignInInner';
 
 export const LoginScreen: FC = () => {
+	const [loading, setLoading] = useState(false);
 	const insets = useSafeAreaInsets();
-	const { authenticationLoading, invitationCode, config } =
-		useSnapshot(appState);
+	const { invitationCode, config } = useSnapshot(appState);
 	const containerStyle = {
 		paddingTop: insets.top,
 		paddingBottom: insets.bottom,
 	};
-
 	const logoSize = 120;
 
+	const handleGoogleSignIn = async () => {
```

Comment:
should not instantiate method this way, if we must manage it under `useCallback`, or put is as file-scoped level function.